cscs-checks/microbenchmarks/fft/src/fftw_benchmark.c

```diff
@@ -0,0 +1,116 @@
+fftw_plan plan_forward_x, plan_backward_x, plan_forward_y, plan_backward_y, plan_forward_z, plan_backward_z;
+
+void init_derivatives(double *func, double *deri, int npx, int npy, int npz, int npy2, int npz2){
+    int nnn;
+    deri_temp_x = (fftw_complex *) malloc(npy*npz*(npx/2+1)*sizeof(fftw_complex));
+    deri_temp_y = (fftw_complex *) malloc(npx*(npy/2+1)*sizeof(fftw_complex));
+    deri_temp_z = (fftw_complex *) malloc(npx*npy2*(npz2/2+1)*sizeof(fftw_complex));
+    nnn = npx;
+    plan_forward_x = fftw_plan_many_dft_r2c(1, &nnn, npy*npz, func, &nnn, 1, npx, deri_temp_x, &nnn, 1, npx/2+1, FFTW_MEASURE+FFTW_UNALIGNED);
+    nnn = npy;
+    plan_forward_y = fftw_plan_many_dft_r2c(1, &nnn, npx, func, &nnn, npx, 1, deri_temp_y, &nnn, 1, npy/2+1, FFTW_MEASURE+FFTW_UNALIGNED);
+    nnn = npz2;
+    plan_forward_z = fftw_plan_many_dft_r2c(1, &nnn, npx*npy2, func, &nnn, npx*npy2, 1, deri_temp_z, &nnn, 1, npz2/2+1, FFTW_MEASURE+FFTW_UNALIGNED);
+    nnn = npx;
+    plan_backward_x = fftw_plan_many_dft_c2r(1, &nnn, npy*npz, deri_temp_x, &nnn, 1, npx/2+1, deri, &nnn, 1, npx, FFTW_MEASURE+FFTW_UNALIGNED);
+    nnn = npy;
+    plan_backward_y = fftw_plan_many_dft_c2r(1, &nnn, npx, deri_temp_y, &nnn, 1, npy/2+1, deri, &nnn, npx, 1, FFTW_MEASURE+FFTW_UNALIGNED);
+    nnn = npz2;
+    plan_backward_z = fftw_plan_many_dft_c2r(1, &nnn, npx*npy2, deri_temp_z, &nnn, 1, npz2/2+1, deri, &nnn, npx*npy2, 1, FFTW_MEASURE+FFTW_UNALIGNED);
+}
+
+void done_derivatives(){
+    fftw_destroy_plan(plan_backward_z);
+    fftw_destroy_plan(plan_backward_y);
+    fftw_destroy_plan(plan_backward_x);
+    fftw_destroy_plan(plan_forward_z);
+    fftw_destroy_plan(plan_forward_y);
+    fftw_destroy_plan(plan_forward_x);
+    free(deri_temp_z);
+    free(deri_temp_y);
+    free(deri_temp_x);
+}
+
+void derivative_x1(double *func, double *deri, int npx, int npy, int npz){
+    int i, jk;
+    fftw_execute_dft_r2c(plan_forward_x, func, deri_temp_x);
+    fftw_execute_dft_c2r(plan_backward_x, deri_temp_x, deri);
+}
+
+void derivative_y1(double *func, double *deri, int npx, int npy, int npz){
+    int i, j, k;
+    for (k = 0; k<npz; k++){
+        fftw_execute_dft_r2c(plan_forward_y, func+k*npy*npx, deri_temp_y);
+        fftw_execute_dft_c2r(plan_backward_y, deri_temp_y, deri+k*npy*npx);
+    }
+}
+
+void derivative_z1(double *func, double *deri, int npx, int npy, int npz){
+    int k, ij;
+    fftw_execute_dft_r2c(plan_forward_z, func, deri_temp_z);
+    fftw_execute_dft_c2r(plan_backward_z, deri_temp_z, deri);
+}
+
+int main(int argc, char *argv[]){
+    int mpi_size, mpi_rank;
+    int npoints, nproc, iter, withmpi;
+    double *fvalue, *dvalue;
+    int npx, npy, npz, npy2, npz2;
+    int i, j, k;
+    double my_time;
+    MPI_Init(&argc, &argv);
+    MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &mpi_size);
+    if (argc != 5){
+        if (mpi_rank == 0){
+            printf("npoints nproc iter\n");
```

Comment:
Print a proper usage message here:

```suggestion
            printf("Usage: %s npoints nproc niter\n", argv[0]);
```